src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -144,17 +146,43 @@ public static class PermutationFeatureImportanceExtensions
                 int? numberOfExamplesToUse = null,
                 int permutationCount = 1) where TModel : class
         {
+            bool isCalibratedModel = false;
+            var type = predictionTransformer.Model.GetType();
+            if (type.IsGenericType)
+            {
+                var genArgs = type.GetGenericArguments();
+                if (Utils.Size(genArgs) == 2)
+                {
+                    var calibratedModelType = typeof(CalibratedModelParametersBase<,>).MakeGenericType(genArgs);
+                    if (calibratedModelType.IsAssignableFrom(type))
+                        isCalibratedModel = true;
+                }
+            }
+            if (isCalibratedModel)
+            {
+                return PermutationFeatureImportance<TModel, BinaryClassificationMetrics, BinaryClassificationMetricsStatistics>.GetImportanceMetricsMatrix(
+                    catalog.GetEnvironment(),
+                    predictionTransformer,
+                    data,
+                    () => new BinaryClassificationMetricsStatistics(),
+                    idv => catalog.Evaluate(idv, labelColumnName),
+                    BinaryClassifierDelta,
+                    predictionTransformer.FeatureColumnName,
+                    permutationCount,
+                    useFeatureWeightFilter,
+                    numberOfExamplesToUse);
+            }
             return PermutationFeatureImportance<TModel, BinaryClassificationMetrics, BinaryClassificationMetricsStatistics>.GetImportanceMetricsMatrix(
-                            catalog.GetEnvironment(),
-                            predictionTransformer,
-                            data,
-                            () => new BinaryClassificationMetricsStatistics(),
-                            idv => catalog.Evaluate(idv, labelColumnName),
-                            BinaryClassifierDelta,
-                            predictionTransformer.FeatureColumnName,
-                            permutationCount,
-                            useFeatureWeightFilter,
-                            numberOfExamplesToUse);
+                catalog.GetEnvironment(),
+                predictionTransformer,
+                data,
+                () => new BinaryClassificationMetricsStatistics(),
+                idv => catalog.EvaluateNonCalibrated(idv, labelColumnName),
```

Comment:
Small Nit:
The only difference between the `if (isCalibratedModel)` and the else case is the idv parameter. Is it possible to make this a bit more readable by factoring out just that line and using a single call to the PermutationFeatureImportance constructor?

Reply:
Resolving this comment, since `isCalibratedModel` has been removed.

---
In reply to: [362685943](https://github.com/dotnet/machinelearning/pull/4587#discussion_r362685943) [](ancestors = 362685943)